### ISSUE TITLE
Segmentation fault 11 when using vw executable

### ISSUE THREAD

**Issue by Mystic98** [VowpalWabbit/vowpal_wabbit#2151]
I'm trying to install vowpalwabbit on macOS Catalina from source. I followed building instruction from https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Building, after using `make` in /vowpal_wabbit, there vw executable appears in /vowpal_wabbit/build/vowpalwabbit. I keep getting SEGFAULT 11 when using it. There is my LastTest.log : 
[LastTest.log](https://github.com/VowpalWabbit/vowpal_wabbit/files/3863797/LastTest.log)
Can something be done with that?

**Comment by peterychang** [comment]
Do you also get segfaults when running the binary outside of the test suite?
If so, I suspect one of the libraries you're linking against isn't binary compatible. Can you try the following?

1. update your package manager to the latest version and package list
2. update your boost and zlib libraries to the latest versions
3. run the following commands from the root directory of vowpalwabbit
```
git clean -xfd
mkdir build
cd build
cmake -DBoost_NO_BOOST_CMAKE=ON ..
make -j `nproc` test_with_output 
```

**Comment by Mystic98** [comment]
Thank you for your answer!
So after `cmake -DBoost_NO_BOOST_CMAKE=ON ..` i got this:

>-- VowpalWabbit Version: 8.7.0
-- Setting build type to 'Release' as none was specified.
-- The CXX compiler identification is AppleClang 11.0.0.11000033
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.21.0 (Apple Git-122.2)") 
-- Git Version: 92aa2d71
-- Number of processors: 4
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found Boost: /usr/local/include (found version "1.71.0") found components:  program_options system thread unit_test_framework chrono date_time atomic 
-- Found ZLIB: /usr/lib/libz.dylib (found version "1.2.11") 
-- Submodule update
CMake Warning (dev) at test/unit_test/CMakeLists.txt:12 (target_sources):
  Policy CMP0076 is not set: target_sources() command converts relative paths
  to absolute.  Run "cmake --help-policy CMP0076" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.
  An interface source of target "vw-unit-test.out" has a relative path.
This warning is for project developers.  Use -Wno-dev to suppress it.
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/evgenich/vowpal_wabbit/build

And after `make -j `nproc` test_with_output` i got this one:

> -bash: nproc: command not found
Scanning dependencies of target allreduce
[  0%] Building CXX object vowpalwabbit/CMakeFiles/allreduce.dir/allreduce_threads.cc.o
[  2%] Building CXX object vowpalwabbit/CMakeFiles/allreduce.dir/allreduce_sockets.cc.o
[  2%] Building CXX object vowpalwabbit/CMakeFiles/allreduce.dir/vw_exception.cc.o
[  3%] Linking CXX static library liballreduce.a
[  3%] Built target allreduce
Scanning dependencies of target vw
[  5%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/accumulate.cc.o
[  5%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/action_score.cc.o
[  5%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/active_cover.cc.o
[  6%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/active.cc.o
[  7%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/audit_regressor.cc.o
[  7%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/autolink.cc.o
[  8%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/baseline.cc.o
[  9%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/best_constant.cc.o
[ 10%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/bfgs.cc.o
[ 10%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/binary.cc.o
[ 11%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/boosting.cc.o
[ 12%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/bs.cc.o
[ 12%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cache.cc.o
[ 13%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cb_adf.cc.o
[ 14%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cb_algs.cc.o
[ 15%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cb_explore_adf_bag.cc.o
[ 15%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cb_explore_adf_regcb.cc.o
[ 16%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cb_explore_adf_first.cc.o
[ 17%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cb_explore_adf_softmax.cc.o
[ 18%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cb_explore_adf_greedy.cc.o
[ 18%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cb_explore_adf_cover.cc.o
[ 19%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cb_explore.cc.o
[ 20%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cb_sample.cc.o
[ 20%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cb.cc.o
[ 21%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cbify.cc.o
[ 22%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/classweight.cc.o
[ 22%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/ccb_label.cc.o
[ 23%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/comp_io.cc.o
[ 25%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/confidence.cc.o
[ 26%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/conditional_contextual_bandit.cc.o
[ 26%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cost_sensitive.cc.o
[ 27%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/cs_active.cc.o
[ 28%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/csoaa.cc.o
[ 28%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/ect.cc.o
[ 29%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/example.cc.o
[ 30%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/explore_eval.cc.o
[ 31%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/ftrl.cc.o
[ 31%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/gd_mf.cc.o
[ 32%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/gd.cc.o
[ 33%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/gen_cs_example.cc.o
[ 33%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/global_data.cc.o
[ 34%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/interact.cc.o
[ 35%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/interactions.cc.o
[ 36%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/io_buf.cc.o
[ 36%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/kernel_svm.cc.o
[ 37%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/label_dictionary.cc.o
[ 37%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/learner.cc.o
[ 38%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/lda_core.cc.o
[ 39%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/log_multi.cc.o
[ 40%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/loss_functions.cc.o
[ 41%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/lrq.cc.o
[ 41%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/lrqfa.cc.o
[ 42%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/marginal.cc.o
[ 43%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/memory_tree.cc.o
[ 43%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/mf.cc.o
[ 44%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/multiclass.cc.o
[ 45%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/multilabel_oaa.cc.o
[ 46%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/multilabel.cc.o
[ 46%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/mwt.cc.o
[ 48%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/network.cc.o
[ 48%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/nn.cc.o
[ 48%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/no_label.cc.o
[ 50%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/noop.cc.o
[ 51%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/oaa.cc.o
[ 52%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/OjaNewton.cc.o
[ 52%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/options_boost_po.cc.o
[ 53%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/options_serializer_boost_po.cc.o
[ 54%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/parse_args.cc.o
[ 55%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/parse_example.cc.o
[ 55%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/parse_primitives.cc.o
[ 56%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/parse_regressor.cc.o
[ 57%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/parser.cc.o
[ 57%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/print.cc.o
[ 58%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/rand48.cc.o
[ 58%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/scorer.cc.o
[ 59%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/recall_tree.cc.o
[ 60%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/search_dep_parser.cc.o
[ 61%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/search_entityrelationtask.cc.o
[ 62%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/search_graph.cc.o
[ 62%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/search_hooktask.cc.o
[ 63%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/search_meta.cc.o
[ 63%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/search_sequencetask.cc.o
[ 64%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/search_multiclasstask.cc.o
[ 65%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/search.cc.o
[ 66%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/sender.cc.o
[ 67%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/shared_feature_merger.cc.o
[ 67%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/simple_label.cc.o
[ 68%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/stagewise_poly.cc.o
[ 69%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/svrg.cc.o
[ 69%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/topk.cc.o
[ 70%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/unique_sort.cc.o
[ 71%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/version.cc.o
[ 72%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/vw_exception.cc.o
[ 72%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/vw_validate.cc.o
[ 73%] Building CXX object vowpalwabbit/CMakeFiles/vw.dir/warm_cb.cc.o
/Users/evgenich/vowpal_wabbit/vowpalwabbit/parser.cc:380:26: warning: 'daemon' is deprecated: first
      deprecated in macOS 10.5 - Use posix_spawn APIs instead. [-Wdeprecated-declarations]
      if (!all.active && daemon(1, 1))
                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/stdlib.h:292:6: note: 
      'daemon' has been explicitly marked deprecated here
int      daemon(int, int) __DARWIN_1050(daemon) __OSX_AVAILABLE_BUT_DEPRECATED_MSG(__MAC_10_...
         ^
/Users/evgenich/vowpal_wabbit/vowpalwabbit/cb_adf.cc:60:72: warning: field '_clip_p' will be
      initialized after field '_no_predict' [-Wreorder]
      : _sd(sd), _model_file_ver(model_file_ver), _rank_all(rank_all), _clip_p(clip_p), _no_...
                                                                       ^
/Users/evgenich/vowpal_wabbit/vowpalwabbit/conditional_contextual_bandit.cc:93:70: warning: missing
      field 'weight' initializer [-Wmissing-field-initializers]
        ex->l.conditional_contextual_bandit.explicit_included_actions});
                                                                     ^
/Users/evgenich/vowpal_wabbit/vowpalwabbit/conditional_contextual_bandit.cc:464:115: warning: 
      missing field 'weight' initializer [-Wmissing-field-initializers]
  ...data.stored_labels[i].outcome, data.stored_labels[i].explicit_included_actions};
                                                                                   ^
/Users/evgenich/vowpal_wabbit/vowpalwabbit/bfgs.cc:117:5: warning: ISO C++11 does not allow
      conversion from string literal to 'char *const' [-Wwritable-strings]
    "Zero or negative curvature detected.\n"
    ^
/Users/evgenich/vowpal_wabbit/vowpalwabbit/conditional_contextual_bandit.cc:464:115: warning: 
      missing field 'weight' initializer [-Wmissing-field-initializers]
/Users/evgenich/vowpal_wabbit/vowpalwabbit/conditional_contextual_bandit.cc:678:32: note: in
      instantiation of function template specialization 'CCB::learn_or_predict<true>' requested here
      init_learner(data, base, learn_or_predict<true>, learn_or_predict<false>, 1, predictio...
                               ^
/Users/evgenich/vowpal_wabbit/vowpalwabbit/conditional_contextual_bandit.cc:464:115: warning: 
      missing field 'weight' initializer [-Wmissing-field-initializers]
  ...data.stored_labels[i].outcome, data.stored_labels[i].explicit_included_actions};
                                                                                   ^
/Users/evgenich/vowpal_wabbit/vowpalwabbit/conditional_contextual_bandit.cc:678:56: note: in
      instantiation of function template specialization 'CCB::learn_or_predict<false>' requested
      here
      init_learner(data, base, learn_or_predict<true>, learn_or_predict<false>, 1, predictio...
                                                       ^
1 warning generated.
4 warnings generated.
1 warning generated.
1 warning generated.
[ 75%] Linking CXX static library libvw.a
[ 75%] Built target vw
Scanning dependencies of target gd_mf_weights
Scanning dependencies of target search_generate
Scanning dependencies of target vw_c_wrapper
Scanning dependencies of target library_example
Scanning dependencies of target ezexample_train
Scanning dependencies of target test_search
Scanning dependencies of target recommend
Scanning dependencies of target vw-bin
Scanning dependencies of target ezexample_predict_threaded
Scanning dependencies of target ezexample_predict
[ 75%] Building CXX object library/CMakeFiles/ezexample_train.dir/ezexample_train.cc.o
[ 76%] Building CXX object library/CMakeFiles/search_generate.dir/search_generate.cc.o
[ 76%] Building CXX object library/CMakeFiles/library_example.dir/library_example.cc.o
[ 76%] Building CXX object library/CMakeFiles/test_search.dir/test_search.cc.o
[ 77%] Building CXX object vowpalwabbit/CMakeFiles/vw_c_wrapper.dir/vwdll.cpp.o
[ 78%] Building CXX object library/CMakeFiles/ezexample_predict.dir/ezexample_predict.cc.o
[ 79%] Building CXX object library/CMakeFiles/gd_mf_weights.dir/gd_mf_weights.cc.o
[ 79%] Building CXX object vowpalwabbit/CMakeFiles/vw-bin.dir/main.cc.o
[ 80%] Building CXX object library/CMakeFiles/recommend.dir/recommend.cc.o
[ 81%] Building CXX object library/CMakeFiles/ezexample_predict_threaded.dir/ezexample_predict_threaded.cc.o
[ 82%] Linking CXX executable library_example
[ 83%] Linking CXX executable ezexample_train
[ 83%] Linking CXX executable ezexample_predict
[ 83%] Built target library_example
[ 83%] Built target ezexample_train
[ 83%] Built target ezexample_predict
[ 84%] Linking CXX executable test_search
[ 84%] Built target test_search
[ 85%] Linking CXX shared library libvw_c_wrapper.dylib
[ 85%] Built target vw_c_wrapper
Scanning dependencies of target vw-unit-test.out
[ 86%] Building CXX object test/unit_test/CMakeFiles/vw-unit-test.out.dir/main.cc.o
[ 87%] Building CXX object test/unit_test/CMakeFiles/vw-unit-test.out.dir/options_test.cc.o
[ 87%] Building CXX object test/unit_test/CMakeFiles/vw-unit-test.out.dir/options_boost_po_test.cc.o
[ 88%] Building CXX object test/unit_test/CMakeFiles/vw-unit-test.out.dir/cb_explore_adf_test.cc.o
[ 89%] Building CXX object test/unit_test/CMakeFiles/vw-unit-test.out.dir/example_header_test.cc.o
[ 89%] Building CXX object test/unit_test/CMakeFiles/vw-unit-test.out.dir/explore_test.cc.o
[ 90%] Building CXX object test/unit_test/CMakeFiles/vw-unit-test.out.dir/stable_unique_tests.cc.o
[ 91%] Building CXX object test/unit_test/CMakeFiles/vw-unit-test.out.dir/object_pool_test.cc.o
[ 92%] Building CXX object test/unit_test/CMakeFiles/vw-unit-test.out.dir/ccb_parser_test.cc.o
[ 92%] Building CXX object test/unit_test/CMakeFiles/vw-unit-test.out.dir/json_parser_test.cc.o
[ 93%] Building CXX object test/unit_test/CMakeFiles/vw-unit-test.out.dir/dsjson_parser_test.cc.o
[ 94%] Building CXX object test/unit_test/CMakeFiles/vw-unit-test.out.dir/ccb_test.cc.o
[ 94%] Building CXX object test/unit_test/CMakeFiles/vw-unit-test.out.dir/vwdll_test.cc.o
[ 95%] Linking CXX executable search_generate
[ 95%] Built target search_generate
[ 96%] Linking CXX executable ezexample_predict_threaded
[ 97%] Linking CXX executable gd_mf_weights
[ 98%] Linking CXX executable vw
[ 98%] Built target ezexample_predict_threaded
[ 98%] Built target gd_mf_weights
[ 98%] Built target vw-bin
[ 98%] Linking CXX executable recommend
[ 98%] Built target recommend
[100%] Linking CXX executable vw-unit-test.out
[100%] Built target vw-unit-test.out
Scanning dependencies of target test_with_output
UpdateCTestConfiguration  from :/Users/evgenich/vowpal_wabbit/build/test/DartConfiguration.tcl
UpdateCTestConfiguration  from :/Users/evgenich/vowpal_wabbit/build/test/DartConfiguration.tcl
Test project /Users/evgenich/vowpal_wabbit/build/test
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 1
    Start 1: RunTests_pass_1_onethread

1: Test command: /usr/local/Cellar/cmake/3.15.5/bin/cmake "-E" "env" "PATH=/Users/evgenich/vowpal_wabbit/build/test/../vowpalwabbit/:/Users/evgenich/env_first.3.7.5/bin:/Users/evgenich/vowpal_wabbit/build/vowpalwabbit:/Library/Frameworks/Python.framework/Versions/3.7/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/opt/X11/bin" "./RunTests" "-d" "-fe" "-E" "0.001" "-O" "--onethread"
1: Test timeout computed to be: 10000000
1: Testing on: hostname=Evgenijs-MacBook-Pro.local OS=darwin
1: Testing vw: /Users/evgenich/vowpal_wabbit/build/test/../vowpalwabbit//vw
1: RunTests: '-D' to see any diff output
1: RunTests: '-o' to force overwrite references
1: sh: line 1: 26941 Segmentation fault: 11  ( /Users/evgenich/vowpal_wabbit/build/test/../vowpalwabbit//vw --onethread -k -l 20 --initial_t 128000 --power_t 1 -d train-sets/0001.dat -f models/0001_1.model -c --passes 8 --invariant --ngram 3 --skips 1 --holdout_off ) > /dev/null 2> 0001.stderr
1: RunTests: test 1: '/Users/evgenich/vowpal_wabbit/build/test/../vowpalwabbit//vw --onethread -k -l 20 --initial_t 128000 --power_t 1 -d train-sets/0001.dat      -f models/0001_1.model -c --passes 8 --invariant      --ngram 3 --skips 1 --holdout_off' failed (exitcode=139)
1: RunTests: exiting with status=139
1/5 Test #1: RunTests_pass_1_onethread ........***Failed    0.53 sec
test 2
    Start 2: RunTests_pass_2

2: Test command: /usr/local/Cellar/cmake/3.15.5/bin/cmake "-E" "env" "PATH=/Users/evgenich/vowpal_wabbit/build/test/../vowpalwabbit/:/Users/evgenich/env_first.3.7.5/bin:/Users/evgenich/vowpal_wabbit/build/vowpalwabbit:/Library/Frameworks/Python.framework/Versions/3.7/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/opt/X11/bin" "./RunTests" "-d" "-fe" "-E" "0.001"
2: Test timeout computed to be: 10000000
2: Testing on: hostname=Evgenijs-MacBook-Pro.local OS=darwin
2: Testing vw: /Users/evgenich/vowpal_wabbit/build/test/../vowpalwabbit//vw
2: RunTests: '-D' to see any diff output
2: RunTests: '-o' to force overwrite references
2: sh: line 1: 26949 Segmentation fault: 11  ( /Users/evgenich/vowpal_wabbit/build/test/../vowpalwabbit//vw -k -l 20 --initial_t 128000 --power_t 1 -d train-sets/0001.dat -f models/0001_1.model -c --passes 8 --invariant --ngram 3 --skips 1 --holdout_off ) > /dev/null 2> 0001.stderr
2: RunTests: test 1: '/Users/evgenich/vowpal_wabbit/build/test/../vowpalwabbit//vw -k -l 20 --initial_t 128000 --power_t 1 -d train-sets/0001.dat      -f models/0001_1.model -c --passes 8 --invariant      --ngram 3 --skips 1 --holdout_off' failed (exitcode=139)
2: RunTests: exiting with status=139
2/5 Test #2: RunTests_pass_2 ..................***Failed    0.07 sec
test 3
    Start 3: save_resume_test

3: Test command: /Users/evgenich/env_first.3.7.5/bin/python "save_resume_test.py" "--verbose_on_fail" "--vw" "/Users/evgenich/vowpal_wabbit/build/test/../vowpalwabbit/vw"
3: Test timeout computed to be: 10000000
3: head: stdout: Broken pipe
3: sh: line 1: 26955 Done(1)                 head -n 107 train-sets/rcv1_small.dat
3:      26956 Segmentation fault: 11  | /Users/evgenich/vowpal_wabbit/build/test/../vowpalwabbit/vw -f tmp.save_resume_test.26951.full --quiet
3/5 Test #3: save_resume_test .................***Failed    0.14 sec
test 4
    Start 4: test_race_condition

4: Test command: /Users/evgenich/env_first.3.7.5/bin/python "repeat.py" "1024" "/Users/evgenich/vowpal_wabbit/build/test/../vowpalwabbit/vw" "train-sets/rcv1_small.dat" "--holdout_after" "100"
4: Test timeout computed to be: 10000000
4: Return code: -11
4: 1 failed out of    2
4/5 Test #4: test_race_condition ..............***Failed    0.07 sec
test 5
    Start 5: vw_unit_test

5: Test command: /Users/evgenich/vowpal_wabbit/build/test/unit_test/vw-unit-test.out
5: Test timeout computed to be: 10000000
5: Running 38 test cases...
5: unknown location:0: fatal error: in "cb_explore_adf_should_throw_empty_multi_example": signal: SIGSEGV, si_code: 0 (memory access violation at address: 0x00000000)
5: /Users/evgenich/vowpal_wabbit/test/unit_test/cb_explore_adf_test.cc:10: last checkpoint: "cb_explore_adf_should_throw_empty_multi_example" test entry
5: 
5: *** 1 failure is detected in the test module "Main"
5: 
5/5 Test #5: vw_unit_test .....................***Failed    0.80 sec

0% tests passed, 5 tests failed out of 5

Total Test time (real) =   1.60 sec

The following tests FAILED:
	  1 - RunTests_pass_1_onethread (Failed)
	  2 - RunTests_pass_2 (Failed)
	  3 - save_resume_test (Failed)
	  4 - test_race_condition (Failed)
	  5 - vw_unit_test (Failed)
Errors while running CTest
make[3]: *** [test/CMakeFiles/test_with_output] Error 8
make[2]: *** [test/CMakeFiles/test_with_output.dir/all] Error 2
make[1]: *** [test/CMakeFiles/test_with_output.dir/rule] Error 2
make: *** [test_with_output] Error 2

**Comment by peterychang** [comment]
@Mystic98 Thanks for the logs. I haven't seen these crashes before, I'll take a look.

A couple of questions on your setup:

- Which version of Catalina are you using?
- Which package manager are you using?
- Are you using any secondary package managers (like Anaconda)?

**Comment by Mystic98** [comment]
@peterychang

-  macOS Catalina 10.15.1

- Homebrew 2.1.16

- No, i didn't use anything else.

**Comment by peterychang** [comment]
Can't repro on OSX 10.14. @ataymano can you repro this on your machine?

**Comment by ataymano** [comment]
reproduced. looking.

**Comment by jackgerrits** [comment]
Fixed in #2152
